src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -99,6 +99,14 @@ def query_imported_fast(self):
         except libzfs.ZFSException as e:
             raise CallError(f'Failed listing imported pools with error: {e}')
 
+    def is_upgraded(self, pool_name):
+        is_upgraded = False
+        with libzfs.ZFS() as zfs:
+            for pool in filter(lambda x: x.name == pool_name, zfs.pools):
+                is_upgraded = not any((i.state.name not in ('ACTIVE', 'ENABLED') for i in pool.features))
```

Comment:
Double negatives are harder to understand. Isn't this `all((i.state.name in ('ACTIVE', 'ENABLED') for i in pool.features)))`

Also is `i.state` an enum? If so, it's better to compare with enum values than with their string representations.